### PR TITLE
Fixed ARM NEON code path in simd.h, all tests in simd_test are passing

### PR DESCRIPTION
**Pull request by gbansal2**
## Description

Fixed ARM NEON code path in src/include/OpenImageIO/simd.h, and a single-line change for a compile error in src/include/OpenImageIO/benchmark.h

## Tests

All tests in simd_test binary are passing

## Checklist:

- [x ] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [ ] If this is more extensive than a small change to existing code, I
  have previously submitted a Contributor License Agreement
  ([individual](../src/doc/CLA-INDIVIDUAL), and if there is any way my
  employers might think my programming belongs to them, then also
  [corporate](../src/doc/CLA-CORPORATE)).
- [ ] I have updated the documentation, if applicable.
- [x ] I have ensured that the change is tested somewhere in the testsuite
  (adding new test cases if necessary).
- [x ] My code follows the prevailing code style of this project.

### CODE REVIEW

**Review by lgritz — COMMENTED** [review]
These LGTM, thanks so much for the fixes!

I think you can remove the commented-out old versions that are incorrect, just replace entirely with the correct code.

File: src/include/OpenImageIO/benchmark.h [hunk 1]
```diff
@@ -26,7 +26,9 @@ OIIO_NAMESPACE_BEGIN
 /// * Google Benchmark https://github.com/google/benchmark/blob/master/include/benchmark/benchmark_api.h
 
 template <class T>
-OIIO_FORCEINLINE T const& DoNotOptimize (T const &val);
+//Do not use always_inline with optnone attribute
+//OIIO_FORCEINLINE T const& DoNotOptimize (T const &val);
```
Comment (lgritz) [inline src/include/OpenImageIO/benchmark.h:30]: you can remove the comments here

File: src/include/OpenImageIO/simd.h [hunk 2]
```diff
@@ -4759,7 +4759,8 @@ OIIO_FORCEINLINE vint4 blend (const vint4& a, const vint4& b, const vbool4& mask
     return _mm_or_si128 (_mm_and_si128(_mm_castps_si128(mask.simd()), b.simd()),
                          _mm_andnot_si128(_mm_castps_si128(mask.simd()), a.simd()));
 #elif OIIO_SIMD_NEON
-    return vbslq_s32 (mask.simd(), a.simd(), b.simd());
+    //return vbslq_s32 (mask.simd(), a.simd(), b.simd());
```
Comment (lgritz) [inline src/include/OpenImageIO/simd.h:4762]: you can remove the comments here

Comment (gbansal2) [reply src/include/OpenImageIO/simd.h]: thanks. removed the comments in the new commit

**Comment by gbansal2** [comment]
I based off my changes from the release branch instead of master. Will that be ok?

**Comment by lgritz** [comment]
Yeah, no prob. The release is only a couple months old, there is not much divergence vs master yet, and as far as I know, none at all in these files. After I merge, I'll just cherry pick the change into master.

**Comment by gbansal2** [comment]
Thank you! For any future changes, I will remember to base off of master.

**Review by lgritz — APPROVED** [review]

**Comment by lgritz** [comment]
I forgot that we keep "release" pointed at the actual latest release tag, so what I've done is added this to RB-2.2 (the tip of what will be the *next* 2.2 release) as well as master.

**Pull request merged**